flutter/lib/app_constants.dart

```diff
@@ -110,5 +110,5 @@ class BackendId {
 }
 
 class BenchmarkSettings {
-  static const double maxDuration = 10 * 60; // 10 minutes
+  static const double maxDuration = 30 * 60; // 30 minutes
```

Comment:
There will be a PR to make this configurable, but for this PR, please keep this unchanged at 10 minutes.